modules/core/src/main/java/org/apache/ignite/internal/processors/cache/GridCacheAdapter.java

```diff
@@ -699,12 +701,12 @@ public void onKernalStop() {
             /*skip values*/true,
             false);
 
-        boolean readRepair = opCtx != null && opCtx.readRepair();
+        ReadRepairStrategy readRepairStrategy = opCtx != null ? opCtx.readRepairStrategy() : null;
 
-        if (readRepair)
+        if (readRepairStrategy != null)
             return getWithRepairAsync(
                 fut,
-                () -> repairAsync(key, opCtx, true),
+                (ks) -> repairAsync(ks, opCtx, true),
```

Comment:
1) This will increase code complexity. 
The same way should be used everywhere.
2) Now we use keys from `IgniteConsistencyViolationException` where keys are `KeyCacheObject's`, while `K key` can be any plain object (that's not a problem now, but can lead to problems in the future).